src/Contract/Transaction.purs

```diff
@@ -405,22 +406,56 @@ unUnbalancedTx
 
 -- | Attempts to balance an `UnbalancedTx` using the specified
 -- | balancer constraints.
-balanceTxWithConstraints
+-- |
+-- | This is a 'non-throwing' variant of this functionality. Use this when
+-- | errors are expected, and graceful recovery is possible. If you want a
+-- | 'throwing' variant, use `balanceTxWithConstraints` instead.
+balanceTxWithConstraintsE
   :: UnbalancedTx
   -> BalanceTxConstraintsBuilder
   -> Contract (Either BalanceTxError.BalanceTxError FinalizedTransaction)
-balanceTxWithConstraints tx =
+balanceTxWithConstraintsE tx =
   let
     tx' /\ ix = unUnbalancedTx tx
   in
     BalanceTx.balanceTxWithConstraints tx' ix
 
--- | Same as `balanceTxWithConstraints`, but uses the default balancer
+-- | 'Throwing' variant of `balanceTxWithConstraintsE`. Instead of returning in
+-- | `Either`, it throws an exception on failure. Use this when errors are not
+-- | expected, and graceful recovery is either impossible or wouldn't make
+-- | sense. If you want a non-'throwing' variant, use `balanceWithConstraintsE`
+-- | instead.
+balanceTxWithConstraints
+  :: UnbalancedTx
+  -> BalanceTxConstraintsBuilder
+  -> Contract FinalizedTransaction
+balanceTxWithConstraints tx bcb = do
+  result <- balanceTxWithConstraintsE tx bcb
+  case result of
+    Left err -> throwError $ error $ BalanceTxError.explainBalanceTxError err
+    Right ftx -> pure ftx
+
+-- | Same as `balanceTxWithConstraintsE`, but uses the default balancer
 -- | constraints.
-balanceTx
+-- |
+-- | This is a 'non-throwing' variant of this functionality. Use this when
+-- | errors are expected, and graceful recovery is possible. If you want a
+-- | 'throwing' variant, use `balanceTx` instead.
```

Comment:
```suggestion
-- | `balanceTx` is a throwing variant of this function.
```

No need for many words. Everyone knows what throwing is. Please adjust the rest in the same manner.